### PR TITLE
[SYCLomatic][CMake] Remove '-static-libstdc++' from target_link_options as '-static-libstdc++' is not supported together with '-fsycl' when using icpx compiler

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

File: clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml [hunk 1]
```diff
@@ -2405,6 +2405,19 @@
       Out: ""
       RuleId: "remove_target_link_options_cuda_arg"
 
+#'-static-libstdc++' is not supported together with '-fsycl' when using icpx compiler.
```
Comment (zhimingwang36) [inline clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml:2428]: please confirm with compiler team, if it is correct or not.

Comment (tomflinda) [reply clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml:2428]: This info is emitted by icpx compiler, when '-static-libstdc++' is used in the linker command:

```
icpx: error: '-static-libstdc++' is not supported with '-fsycl'

```

**Review by zhiweij1 — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]

File: clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml [hunk 2]
```diff
@@ -2405,6 +2405,19 @@
...
+- Rule: rule_libstdc++_target_link_options
+  Kind: CMakeRule
+  Priority: Fallback
+  CmakeSyntax: libstdc++_target_link_options
+  In: target_link_options${empty}(${target} ${libs})
```
Comment (zhimingwang36) [inline clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml:2433]: the rule doesn't distinguish whether gcc/g++/icpx -fsycl is used to build.  So please emit warrning as only -fsycl doen't support -static-libstd++

Comment (tomflinda) [reply clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml:2433]: Will implement it in a separate pr.

**Review by ziranzha — APPROVED** [review]

**Pull request merged**